### PR TITLE
tests: add all tests to GitHub workflow

### PR DESCRIPTION
**Pull request by connorjclark**
Let's inch closer to using GH for our main CI.

I expect all the tests that use Chrome to fail. Need to set up a docker image w/ Chrome installed.

### CODE REVIEW

**Comment by connorjclark** [comment]
Seems like there isn't a way to keep running steps even if one fails. There is this https://help.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idcontinue-on-error but it would pass the build. @paulirish did you look into this before?

**Comment by paulirish** [comment]
> There is this [help.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idcontinue-on-error](https://help.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idcontinue-on-error) but it would pass the build. @paulirish did you look into this before?

i thought i did, but i was wrong. i was adding https://help.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idstrategyfail-fast but it affects things at a JOB level, not at a JOB STEP level.

**Comment by connorjclark** [comment]
paul wants to follow up with job parallelization: https://help.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idneeds

**Comment by connorjclark** [comment]
this is g2g

**Comment by connorjclark** [comment]
we want to:

* remove bundle size (we dont want it)
* land this and #10622
* figure out how to run all steps even if one fails, but still fail the build if one did fail. some combination of `continue-on-error` and `if: failure()`?
* job parallelization
* keep travis, appveyor around for some time
* no rush but to remove other CIs: add coverage upload, windows, different node versions

File: .github/workflows/basics.yml [hunk 1]
```diff
@@ -40,9 +39,22 @@ jobs:
     - run: yarn --frozen-lockfile
     - run: yarn build-all
     - run: yarn diff:sample-json
+    - run: yarn type-check
     - run: yarn lint
     - run: yarn unit-core
```
Comment (paulirish) [inline .github/workflows/basics.yml:44]: also need `unit-cli` and `unit-viewer`

i added a note that these will move to their coverage eequivalents later.

**Pull request merged**